From 0d48c3031989e38353fc49e94e6575dd27342457 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Thu, 05 Aug 2021 08:26:36 +0300
Subject: [PATCH] [SW-50869] arch: move hl_rst_src to platform specific headers

Gaudi1 and Greco has common bit definitions in PSOC_GLOBAL_CONF_RST_SRC
register for reset cause. But in Gaudi2 reset cause bit definitions are
modified and not backward compatible.

Moved hl_rst_src from common include header to platform specific header
files. Gadui2 specific enum changes will be done later

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I1421f9a698f3fc82d99eea69d5398e80e7415f7d
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 7b4051f..1c22e89 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -21,6 +21,7 @@
 #include "pll.h"
 #include <boot_if.h>
 #include <pci.h>
+#include <hl.h>
 
 __weak bool ppboot_is_valid_img_desc(struct boot_desc *desc)
 {
diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index d0afa06..bf9e946 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -33,6 +33,7 @@
 #include "hl_i2c_slave_events.h"
 #include "rma_info.h"
 #include <spi.h>
+#include <hl.h>
 
 struct hl_rsp_err {
 	union {
diff --git a/arch/gaudi/arm/asic.c b/arch/gaudi/arm/asic.c
index bd8cd01..262214d 100644
--- a/arch/gaudi/arm/asic.c
+++ b/arch/gaudi/arm/asic.c
@@ -66,6 +66,7 @@
 #include <address_space.h>
 
 #include "common.h"
+#include <hl.h>
 
 u32 hl_skip_pcie_init(void)
 {
diff --git a/arch/gaudi/arm/hl.h b/arch/gaudi/arm/hl.h
new file mode 100644
index 0000000..85aaba6
--- /dev/null
+++ b/arch/gaudi/arm/hl.h
@@ -0,0 +1,17 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2021 HabanaLabs Ltd.
+ */
+
+#ifndef ARCH_GAUDI_ARM_HL_H_
+#define ARCH_GAUDI_ARM_HL_H_
+
+enum hl_rst_src {
+	HL_COLD_RST,
+	HL_MANUAL_RST,
+	HL_PRSTN_RST = 2,
+	HL_WD_RST = 4,
+	HL_SW_ALL_RST = 8,
+};
+
+#endif /* ARCH_GAUDI_ARM_HL_H_ */
diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index 6ec54d8..f942998 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -14,6 +14,7 @@
 #include <uart.h>
 
 #include "common.h"
+#include <hl.h>
 
 u32 hl_skip_pcie_init(void)
 {
diff --git a/arch/gaudi2/common/hl.h b/arch/gaudi2/common/hl.h
index e46ee24..a2f7240 100644
--- a/arch/gaudi2/common/hl.h
+++ b/arch/gaudi2/common/hl.h
@@ -6,6 +6,14 @@
 #ifndef ARCH_GAUDI2_COMMON_HL_H_
 #define ARCH_GAUDI2_COMMON_HL_H_
 
+enum hl_rst_src {
+	HL_COLD_RST,
+	HL_MANUAL_RST,
+	HL_PRSTN_RST = 2,
+	HL_WD_RST = 4,
+	HL_SW_ALL_RST = 8,
+};
+
 void set_pci_msix_pba_offset_val(void);
 void set_slv_cfg_reg_val(void);
 
diff --git a/arch/goya/arm/hl.h b/arch/goya/arm/hl.h
new file mode 100644
index 0000000..6c5c535
--- /dev/null
+++ b/arch/goya/arm/hl.h
@@ -0,0 +1,17 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2021 HabanaLabs Ltd.
+ */
+
+#ifndef ARCH_GOYA_ARM_HL_H_
+#define ARCH_GOYA_ARM_HL_H_
+
+enum hl_rst_src {
+	HL_COLD_RST,
+	HL_MANUAL_RST,
+	HL_PRSTN_RST = 2,
+	HL_WD_RST = 4,
+	HL_SW_ALL_RST = 8,
+};
+
+#endif /* ARCH_GOYA_ARM_HL_H_ */
diff --git a/arch/greco/arc_elf32/asic.c b/arch/greco/arc_elf32/asic.c
index bcb4aba..133e0f3 100644
--- a/arch/greco/arc_elf32/asic.c
+++ b/arch/greco/arc_elf32/asic.c
@@ -10,6 +10,7 @@
 #include <preboot.h>
 
 #include "common.h"
+#include <hl.h>
 
 /*
  * FW_SRAM layout:
diff --git a/arch/greco/common/hl.h b/arch/greco/common/hl.h
index 21ed3d8..e441440 100644
--- a/arch/greco/common/hl.h
+++ b/arch/greco/common/hl.h
@@ -40,6 +40,14 @@
 #define ARC0_MEM_SIZE			(ARC0_ICCM_SIZE + ARC0_DCCM_SIZE)
 #define ARC1_MEM_SIZE			(ARC1_ICCM_SIZE + ARC1_DCCM_SIZE)
 
+enum hl_rst_src {
+	HL_COLD_RST,
+	HL_MANUAL_RST,
+	HL_PRSTN_RST = 2,
+	HL_WD_RST = 4,
+	HL_SW_ALL_RST = 8,
+};
+
 void run_arc(u8 id);
 void halt_n_reset_arc(u8 id);
 void axi_enable_drain(void);
diff --git a/include/preboot.h b/include/preboot.h
index 7acc655..94c7b52 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -56,14 +56,6 @@
 #define __unused	__attribute__((__unused__))
 #endif
 
-enum hl_rst_src {
-	HL_COLD_RST,
-	HL_MANUAL_RST,
-	HL_PRSTN_RST = 2,
-	HL_WD_RST = 4,
-	HL_SW_ALL_RST = 8,
-};
-
 enum hl_debug_hw_func {
 	HL_DEBUG_HW_PCIE,
 	HL_DEBUG_HW_DRAM,
diff --git a/lib/preboot/module_pci.c b/lib/preboot/module_pci.c
index 7653b25..1ecfaa4 100644
--- a/lib/preboot/module_pci.c
+++ b/lib/preboot/module_pci.c
@@ -16,6 +16,7 @@
 #include "uart.h"
 #include "pci.h"
 #include "pll.h"
+#include <hl.h>
 
 struct pci_status {
 	u32 flr_reg_val;
